src/RageSoundMixBuffer.cpp

```diff
@@ -1,22 +1,38 @@
 #include "global.h"
 #include "RageSoundMixBuffer.h"
 #include "RageUtil.h"
+#include "PrefsManager.h"
+#include "RageLog.h"
 
 #include <cmath>
 #include <cstdint>
 #include <cstdlib>
 
+
+/* NOTE(sukibaby): kBufferSizeMid is a multiple of the LCM of 44100 and 48000.
+ * This is to ensure a smooth experience regardless of which sample rate is used.
+ * I am populating the default kAdjustedBufferSize with the mid buffer size,
+ * so we don't crash if the preference is invalid or missing. */
+static unsigned GetBufferSizeFromPreference();
+constexpr unsigned kBufferSizeMid = 35280;
+constexpr unsigned kBufferSizeLow = kBufferSizeMid / 2;
+constexpr unsigned kBufferSizeHigh = kBufferSizeMid * 2;
+constexpr unsigned kBufferSizeHuge = kBufferSizeMid * 8;
+static size_t kAdjustedBufferSize = sizeof(float) * kBufferSizeMid;
+
+
 RageSoundMixBuffer::RageSoundMixBuffer()
 {
-	m_pMixbuf = static_cast<float*>(std::malloc(BUF_SIZE * sizeof(float)));
-	if (m_pMixbuf == nullptr)
-	{
-		ASSERT_M(false, "Failed to allocate memory for the sound mixing buffer");
-	}
-	m_iBufSize = BUF_SIZE;
-	std::memset(m_pMixbuf, 0, m_iBufSize * sizeof(float));
+	static const unsigned kMixBufferSize = static_cast<size_t>(GetBufferSizeFromPreference());
```

Comment:
Silly oversight here to statically cast to `size_t` then store in an `unsigned`, but I'll take care of it on the next push, as it's not critical.